examples/quickstart/jupyter-notebooks/notebooks/03-query/02-approxRanking.ipynb

```diff
@@ -0,0 +1,577 @@
+    "\n",
+    "Notice the `threshold` value?\n",
+    "\n",
+    "```json\n",
+    "    \"threshold\": 10,\n",
+    "```\n",
+    "\n",
+    "The parallelised `LIMIT` was the `max` of both the `threshold` shown here – which came from the `LIMIT` in the SQL - and a configuration setting in your cluster – the default for which is 1,000.\n",
+    "\n",
+    "You can find out how to read and set this default `LIMIT` in the [documentation](https://druid.apache.org/docs/latest/querying/topnquery.html#aliasing).\n",
+    "\n",
+    "As a first step in understanding the implications, we need to find data in our sample set where the\n",
+    "cardinality of the dimension that we will `GROUP BY` exceeds that number. By default, that is 1000.\n",
+    "\n",
+    "What's the cardinality of our dimension?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "65a968e5-d51e-47e9-af04-88181f3b865b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql = '''\n",
+    "SELECT COUNT (DISTINCT \"Reporting_Airline\") AS UniqueReportingAirlines\n",
+    "FROM \"example-flights-topn\"\n",
+    "'''\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "13c8a101-dcba-49be-8d05-0a5dbd9731ca",
+   "metadata": {},
+   "source": [
+    "Twenty unique values is too low – the initial `LIMIT` has no effect! This means there is no trimming happening anywhere in the database. As a result, as the documentation explains, our results are going to be without error. All the data servers will return all their results, without trimming, to be merged and passed back to us.\n",
+    "\n",
+    "Let's find another dimension."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "71c1816b-f090-4a3d-b476-8d40eb9c2dec",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql = '''\n",
+    "SELECT\n",
+    "    COUNT (DISTINCT \"Tail_Number\") AS UniqueTailNumbers\n",
+    "FROM \"example-flights-topn\"\n",
+    "WHERE \"Tail_Number\" <> ''\n",
+    "'''\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "72a1a858-bda8-464e-988b-c4ed80b63f43",
+   "metadata": {},
+   "source": [
+    "With this many distinct values to `GROUP BY`, we know that data servers will trim their results when the\n",
+    "`topN` engine is engaged.\n",
+    "\n",
+    "Run the next query to visualise the distribution of unique `Tail_Number`s in the example dataset."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "208f1463-34dd-4b0e-aa78-e582e2133a8f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql = '''\n",
+    "SELECT\n",
+    "    \"Tail_Number\",\n",
+    "    COUNT(*) AS RecordCount\n",
+    "FROM \"example-flights-topn\"\n",
+    "WHERE \"Tail_Number\" <> ''\n",
+    "GROUP BY 1\n",
+    "ORDER BY 2 DESC\n",
+    "LIMIT 500\n",
+    "'''\n",
+    "\n",
+    "df4 = pd.DataFrame(sql_client.sql(sql))\n",
```

Comment:
suggest adding simple comments within the script to say what it's doing. For example
```
# Load the results into a pandas DataFrame and 
# Display the result in a simple line graph [link to pandas plot doc?]
```

Reply:
Cool - will do that.